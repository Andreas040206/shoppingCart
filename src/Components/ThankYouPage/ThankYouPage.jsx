import ThankYouPageStyles from "./ThankYouPage.module.css";
import { Button } from "../Button/Button.jsx";
import { useState } from "react";
import PropTypes from "prop-types";
import { useNavigate } from "react-router-dom";

export const ThankYouPage = ({ setIsThankPage, setIsClosingShoppingBag }) => {
  const [isClosing, setIsClosing] = useState(false);
  const navigate = useNavigate();

  return (
    <div
      className={
        ThankYouPageStyles.bgCon +
        // Making sure the animation is executed
        " " +
        (isClosing === true ? ThankYouPageStyles.bgConClosing : null)
      }
    >
      <div
        className={
          ThankYouPageStyles.con +
          // Making sure the animation is executed
          " " +
          (isClosing === true ? ThankYouPageStyles.conClosing : null)
        }
      >
        <h1 className={ThankYouPageStyles.header}>Thank you!</h1>
        <span className={ThankYouPageStyles.text}>
          Thanks for testing my webside
        </span>
        <Button
          text={"Try again!"}
          fn={() => {
            setIsClosing(true);
            setIsClosingShoppingBag(true);
            setTimeout(() => {
              // It shouldn't remove the element before the animation
              setIsThankPage(false);
              navigate("/", { replace: true });
            }, 1300);
          }}
        />
      </div>
    </div>
  );
};

ThankYouPage.propTypes = {
  setIsThankPage: PropTypes.func.isRequired,
};
