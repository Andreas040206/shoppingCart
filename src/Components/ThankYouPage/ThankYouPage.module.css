.bgCon {
  position: fixed;
  height: 100vh;
  width: 100vw;
  top: 0px;
  left: 0px;
  background-color: rgba(0, 0, 0, 0.39);
  z-index: 100;
  animation: 0.3s bgFade;
  display: flex;
  justify-content: center;
  align-items: center;
}

.bgConClosing {
  animation: bgFadeOut 0.3s;
  animation-delay: 1s;
}

@keyframes bgFade {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@keyframes bgFadeOut {
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}

.con {
  height: min(400px, 95vh);
  width: min(600px, 95vw);
  background-color: #fff;
  animation: conFadeIn 1s ease-out;
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.conClosing {
  animation: conFadeOut 1s forwards ease-in;
}

@keyframes conFadeIn {
  0% {
    opacity: 0;
    transform: translateY(-400px);
  }
  40% {
    opacity: 0;
  }
  80% {
    opacity: 1;
  }
  100% {
    transform: translateY(0px);
    opacity: 1;
  }
}

@keyframes conFadeOut {
  0% {
    opacity: 1;
    transform: translateY(0px);
  }
  20% {
    opacity: 1;
  }
  60% {
    opacity: 0;
  }
  100% {
    transform: translateY(400px);
    opacity: 0;
  }
}

.header {
  font-size: 40px;
  margin: 0px 0px 10px 0px;
}

.text {
  color: #424242;
  margin-bottom: 50px;
}
