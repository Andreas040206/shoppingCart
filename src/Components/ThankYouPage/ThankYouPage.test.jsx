import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";

import { ThankYouPage } from "./ThankYouPage";

describe("Testing ThankYouPage component", () => {
  it("Loads correctly", () => {
    const { container } = render(<ThankYouPage />);
    expect(container).toMatchSnapshot();
  });
});
