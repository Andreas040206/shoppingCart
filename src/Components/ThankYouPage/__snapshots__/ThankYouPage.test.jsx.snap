// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing ThankYouPage component > Loads correctly 1`] = `
<div>
  <div
    class="_bgCon_96ca8e null"
  >
    <div
      class="_con_96ca8e null"
    >
      <h1
        class="_header_96ca8e"
      >
        Thank you!
      </h1>
      <span
        class="_text_96ca8e"
      >
        Thanks for testing my webside
      </span>
      <button
        class="_btn_04cbf6 _dark_04cbf6"
      >
        Try again!
      </button>
    </div>
  </div>
</div>
`;
